src/rules/no-non-numeric-dimensions/index.js

```diff
@@ -0,0 +1,127 @@
+import { utils } from "stylelint";
+import { namespace } from "../../utils";
+import valueParser from "postcss-value-parser";
+
+export const ruleName = namespace("no-non-numeric-dimensions");
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: unit => `Expected "$value * 1${unit}" instead of "#{value}${unit}"`
```

Comment:
Consider also suggesting that they write `$value` in terms of `unit` to begin with. While `$value * 1${unit}` will mostly work, it's likely that their overall logic will be more coherent if they start out using the correct units.